src/PrestaShopBundle/Service/Database/DoctrineFolderSchemaUpdater.php

```diff
@@ -0,0 +1,283 @@
+        $removalSqlStatements = $this->schemaTool->getUpdateSchemaSql($remainingClassList, false);
+        $protectedTables = $this->getProtectedTables($folderClassList);
+
+        return $this->filterSqlStatements($removalSqlStatements, $protectedTables);
+    }
+
+    /**
+     * Removes database schema based on entities' folder, using doctrine schema update.
+     * It only removes tables related to the provided folder.
+     *
+     * @param string $entitiesFolder
+     *
+     * @throws DBALException
+     * @throws EntityFolderNotFoundException
+     * @throws EntityNamespaceNotFoundException
+     */
+    public function removeFolderSchema(string $entitiesFolder)
+    {
+        $sqlStatements = $this->getFolderSchemaRemoval($entitiesFolder);
+
+        $this->executeStatements($sqlStatements);
+    }
+
+    /**
+     * @param string $entitiesFolder
+     *
+     * @return ClassMetadata[]
+     *
+     * @throws EntityFolderNotFoundException
+     * @throws EntityNamespaceNotFoundException
+     */
+    private function getFolderClassList(string $entitiesFolder): array
+    {
+        if (!is_dir($entitiesFolder)) {
+            throw new EntityFolderNotFoundException(sprintf(
+                'Cannot find entity folder %s',
+                $entitiesFolder
+            ));
+        }
+
+        $entitiesNamespace = $this->getFolderNamespace($entitiesFolder);
+
+        return $this->getMetadataListByNamespace($entitiesNamespace);
+    }
+
+    /**
+     * @param string $entitiesNamespace
+     *
+     * @return ClassMetadata[]
+     */
+    private function getMetadataListByNamespace(string $entitiesNamespace): array
+    {
+        $metadataList = [];
+        /** @var ClassMetadata $metadata */
+        foreach ($this->em->getMetadataFactory()->getAllMetadata() as $metadata) {
+            if ($metadata->namespace === $entitiesNamespace) {
+                $metadataList[] = $metadata;
+            }
+        }
+
+        return $metadataList;
+    }
+
+    /**
+     * @param string $entitiesFolder
+     *
+     * @return string
+     *
+     * @throws EntityNamespaceNotFoundException
+     */
+    private function getFolderNamespace(string $entitiesFolder)
+    {
+        $finder = new Finder();
+        $finder->files()->in($entitiesFolder)->name('*.php');
+        foreach ($finder as $phpFile) {
+            $phpContent = file_get_contents($phpFile->getRealPath());
+            if (preg_match('~namespace[ \t]+(.+)[ \t]*;~Um', $phpContent, $matches)) {
+                return $matches[1];
+            }
+        }
+
+        throw new EntityNamespaceNotFoundException(sprintf(
+            'Cannot find namespace in folder %s',
+            $entitiesFolder
+        ));
+    }
+
+    /**
+     * @param array $sqlStatements
+     * @param array $protectedTables
+     *
+     * @return array
+     */
+    private function filterSqlStatements(array $sqlStatements, array $protectedTables): array
```

Comment:
Could you explain why these functions are needed?

Reply:
The purpose is to make no modifications on other tables, to avoid removing/altering core tables (or other modules tables) This was inspired by our custom `UpdateSchemaCommand`
However there is probably room for improvement, if you have ideas I'm interested, in the meantime I will add a comment to explain the purpose of this filtering